server/build/plugins/pages-plugin.js

```diff
@@ -26,6 +26,8 @@ export default class PagesPlugin {
           routeName = routeName.replace(/\\/g, '/')
         }
 
+        routeName = `/${routeName.replace(/(^|\/)index$/, '')}`
```

Comment:
Before this receiced `abcindex` and produced `/abc`
Now it only matches `index` or `abc/index`
And it does that after the windows slash conversion